projects/PeerTube/test.nix

```diff
@@ -0,0 +1,172 @@
+{
+  sources,
+  pkgs,
+  lib,
+  ...
+}: let
+  storageBase = "/var/peertube";
+  storageDir = subdir: "${storageBase}/${subdir}/";
+  peerUser = "peertube";
+  peerGroup = "peertube";
+  localUser = "alice";
+  localDomain = "localhost";
+  listenWeb = 9000;
+  pluginPkgs = with pkgs; [
+    peertube-plugin-akismet
+    peertube-plugin-auth-ldap
+    peertube-plugin-auth-openid-connect
+    peertube-plugin-auth-saml2
+    peertube-plugin-auto-block-videos
+    peertube-plugin-auto-mute
+    peertube-plugin-hello-world
+    peertube-plugin-logo-framasoft
+    peertube-plugin-matomo
+    peertube-plugin-privacy-remover
+    peertube-plugin-transcoding-custom-quality
+    peertube-plugin-transcoding-profile-debug
+    peertube-plugin-video-annotation
+    peertube-theme-background-red
+    peertube-theme-dark
+    peertube-theme-framasoft
+
+    peertube-plugin-livechat
+  ];
+in {
+  name = "peertube-plugins";
+
+  nodes = {
+    server = {config, ...}: {
+      imports = [
+        sources.modules.default
+        sources.modules."services.peertube.plugins"
+      ];
+
+      # Interactive testing can run into OOM with default
+      virtualisation.memorySize = 2047;
+
+      users.users."${localUser}" = {
+        description = localUser;
+        password = "foobar";
+        isNormalUser = true;
+        extraGroups = ["wheel"];
+        uid = 1000;
+      };
+
+      services.xserver = {
+        enable = true;
+        displayManager.lightdm.enable = true;
+        windowManager.icewm.enable = true;
+      };
+
+      services.displayManager = {
+        defaultSession = "none+icewm";
+        autoLogin = {
+          enable = true;
+          user = localUser;
+        };
+      };
+
+      environment = {
+        # To ease with interactive testing, use a fixed password
+        etc."peertube-envvars".text = ''
+          PT_INITIAL_ROOT_PASSWORD=changeme
+        '';
+      };
+
+      services.peertube = {
```

Comment:
Please extract the configuration under test into a separate file that can be exposed as an example

Reply:
https://github.com/ngi-nix/ngipkgs/pull/306/commits/14dcedd1b597f8c6d5ba1de2710929db386e6cad#diff-59884480f9a83e10a7255d516e55d880d2d286e2d5134c1238294f4d35fb206c